torch/testing/_internal/distributed/multi_threaded_pg.py

```diff
@@ -232,6 +232,7 @@ class ProcessLocalGroup(dist.ProcessGroup):
     _pg_list = []
     _count = 0
     _ready = False
```

Comment:
can we remove the `_ready` flag too? the whole point of this flag can be substitute by the ready_cond cv.

Reply:
Yeah we had `_ready` to serve the same purpose but somehow the usage was removed....

Reply:
I attempted to remove this flag. It turned out to expose some issues in readiness check.
@wanchaol @XilunWu please check my latest change in this PR as I removed the class vars related to readiness check and simply used store based barrier instead (see inline comment for rationale).

The reason why store based barrier instead of regular barrier is because latter would have chicken-n-egg problem otherwise (cannot execute collective before pg is ready) as the current threaded pg doesn't have its own rendezvous like gloo or nccl.